cloudbot/client.py

```diff
@@ -55,11 +69,33 @@ def __init__(self, bot, name, nick, *, channels=None, config=None):
         # set when on_load in core_misc is done
         self.ready = False
 
+        self._active = False
+
     def describe_server(self):
         raise NotImplementedError
 
     @asyncio.coroutine
-    def connect(self):
+    def auto_reconnect(self):
+        if not self._active:
+            return
+
+        yield from self.try_connect()
+
+    @asyncio.coroutine
+    def try_connect(self):
+        timeout = 30
+        while not self.connected:
+            try:
+                yield from self.connect(timeout)
+            except Exception:
+                logger.exception("[%s] Error occurred while connecting.", self.name)
```

Comment:
There are a number of instances of using this string formatting instead of ```.format```. i'd recommend changing it.